app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/OffboardRouterActivityJava.java

```diff
@@ -0,0 +1,224 @@
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_mock_navigation);
+    ButterKnife.bind(this);
+
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @OnClick(R.id.newLocationFab)
+  public void onNewLocationClick() {
+    newOrigin();
+  }
+
+  private void newOrigin() {
+    if (mapboxMap != null) {
+      clearMap();
+      LatLng latLng = Utils.getRandomLatLng(new double[]{-77.1825, 38.7825, -76.9790, 39.0157});
+      origin = Point.fromLngLat(latLng.getLongitude(), latLng.getLatitude());
+      mapboxMap.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 12));
+    }
+  }
+
+  @SuppressLint("MissingPermission")
+  @Override
+  public void onMapReady(@NonNull MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    this.mapboxMap.addOnMapClickListener(this);
+    mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+      navigationMapRoute = new NavigationMapRoute(mapView, mapboxMap);
+      Snackbar.make(findViewById(R.id.container), "Tap map to place waypoint", Snackbar.LENGTH_LONG).show();
+      newOrigin();
+    });
+  }
+
+  @Override
+  public boolean onMapClick(@NonNull LatLng point) {
+    if (destination == null) {
+      destination = Point.fromLngLat(point.getLongitude(), point.getLatitude());
+      mapboxMap.addMarker(new MarkerOptions().position(point));
+      findRoute();
+    } else if (waypoint == null) {
+      waypoint = Point.fromLngLat(point.getLongitude(), point.getLatitude());
+      mapboxMap.addMarker(new MarkerOptions().position(point));
+      findRoute();
+    } else {
+      Toast.makeText(this, "Only 2 waypoints supported", Toast.LENGTH_LONG).show();
+      clearMap();
+    }
+    return false;
+  }
+
+  private void clearMap() {
+    if (mapboxMap != null) {
+      mapboxMap.clear();
+      route = null;
+      destination = null;
+      waypoint = null;
+      navigationMapRoute.updateRouteVisibilityTo(false);
+      navigationMapRoute.updateRouteArrowVisibilityTo(false);
+    }
+  }
+
+  private void findRoute() {
+    if (directionsSession == null) {
+      Router offboardRouter = new MapboxOffboardRouter(
+        this,
+        Utils.getMapboxAccessToken(this)
+      );
+      directionsSession = new MapboxDirectionsSession(
+        offboardRouter,
+        this
+      );
+    }
+    if (origin != null && destination != null) {
+      if (TurfMeasurement.distance(origin, destination, TurfConstants.UNIT_METERS) > 50) {
+        List<Point> waypoints = new ArrayList<>();
+        if (waypoint != null) {
+          waypoints.add(waypoint);
+        }
+        directionsSession.requestRoutes(
+          origin,
+          waypoints,
+          destination
+        );
+      }
+    }
+  }
+
+  /*
+   * DirectionSessions.RouteObserver
+   */
+
+  @Override
+  public void onRoutesChanged(@NotNull List<Route> routes) {
+    if (!routes.isEmpty()) {
+      route = MappersKt.mapToDirectionsRoute(routes.get(0));
```

Comment:
We should add `@file:JvmName("Mappers")` into `Mappers.kt` so Java clients don't need to type `Kt` 😉